### DISCUSSION TITLE
Installing Deye SUN800G3 Grid Inverter in Home Assistant

### DISCUSSION THREAD

**Discussion by fluchderpixel** [StephanJoubert/home_assistant_solarman#325] [Show and tell]
Hello,

after a long search and many hours of trying things out, I would now like to share my findings here to save others the hassle. The general installation is done according to the instructions (Hacs -->Integration-->Solarman-->restart Home Assistant). Then copy the content of deye_4mppt.yaml into a custom_parameters.yaml and change the following part:

```python
    - name: "Total Production"
      class: "energy"
      state_class: "total_increasing"
      uom: "kWh"
      scale: 0.1
      rule: 3
      registers: [0x003F,0x0040]
      icon: 'mdi:solar-power'
#        validation:
#        min: 0.1
#        invalidate_all:
``` 

Save and then continue according to the instructions in Home Assistant. Select the custom_parameters.yaml as the lookup file.
After saving, it took me only a few minutes and the data was displayed in Home Assistant.

I hope to be able to support others with this procedure.

Thanks also to the developers :-)

**Comment by hagagit** [comment]
I think this inverter is a 2 MPPT version. There is a deye_2mppt.yaml available. This works for me out of the box on a similar inverter.

I think a new release would be great, so people get this out of the box, without big hassle.

**Reply by fluchderpixel** [reply]
Yes, of course, it is an inverter with 2 MPPT interfaces 😄.

However, the deye_2mppt.yaml is not yet available in the standard installation. During the manual installation of the deye_2mppt.yaml, it did not work either. Only after the validation passage was also commented out, did Home Assistant display the inverter data.

If the deye_2mppt.yaml is available in the official installation, I will change the text accordingly 🤓.

I don't know what task the validation part has in "total production". It would not have worked with my Deye SUN800G3-EU-230.
It would be even better if the validation part was commented out beforehand. However, I don't have the right knowledge to add this to GitHub, so a programmer would have to check what other effects the change has.
